web/apps/serve.go

```diff
@@ -302,6 +304,25 @@ func BuildTemplates() {
 	themeTemplate = template.Must(template.New("theme").Funcs(middlewares.FuncsMap).Parse(`
 <link rel="stylesheet" type="text/css" href="{{asset .Domain "/styles/theme.css" .ContextName}}">
 	`))
+
+	faviconTemplate = template.Must(template.New("favicon").Funcs(middlewares.FuncsMap).Parse(`
+<link rel="icon" href="{{asset .Domain "/favicon.ico" .ContextName}}">
+<link rel="icon" type="image/png" href="{{asset .Domain "/favicon-16x16.png" .ContextName}}" sizes="16x16">
+<link rel="icon" type="image/png" href="{{asset .Domain "/favicon-32x32.png" .ContextName}}" sizes="32x32">
+<link rel="apple-touch-icon" sizes="180x180" href="{{asset .Domain "/apple-touch-icon.png" .ContextName}}"/>
+	`))
+}
+
+func favicon(i *instance.Instance) template.HTML {
+	buf := new(bytes.Buffer)
+	err := faviconTemplate.Execute(buf, echo.Map{
+		"Domain":      i.ContextualDomain(),
+		"ContextName": i.ContextName,
+	})
+	if err != nil {
+		panic(err)
+	}
+	return template.HTML(buf.String()) // #nosec
```

Comment:
Do you think it would be complicated to move `favicon` and `themecss` to `web/middlewares`, so that we can have the same helpers for both the stack assets and the webapps pages?

Reply:
Yep, nice pick ;)